test/benchmarks_suite.go

```diff
@@ -0,0 +1,115 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	pstore "github.com/libp2p/go-libp2p-peerstore"
+)
+
+var peerstoreBenchmarks = map[string]func(pstore.Peerstore, chan *peerpair) func(*testing.B){
+	"AddAddrs": benchmarkAddAddrs,
+	"SetAddrs": benchmarkSetAddrs,
+	"GetAddrs": benchmarkGetAddrs,
+	// The in-between get allows us to benchmark the read-through cache.
+	"AddGetAndClearAddrs": benchmarkAddGetAndClearAddrs,
+	// Calls PeersWithAddr on a peerstore with 1000 peers.
+	"Get1000PeersWithAddrs": benchmarkGet1000PeersWithAddrs,
+}
+
+func BenchmarkPeerstore(b *testing.B, factory PeerstoreFactory, variant string) {
+	// Parameterises benchmarks to tackle peers with 1, 10, 100 multiaddrs.
+	params := []struct {
```

Comment:
We now run parameterised benchmarks to characterise the performance of operations with peers with 1 multiaddr, 10 multiaddr, 100 multiaddr.